library/core/src/ptr/const_ptr.rs

```diff
@@ -38,32 +38,30 @@ impl<T: ?Sized> *const T {
         self as _
     }
 
-    /// Returns `None` if the pointer is null, or else returns a reference to
+    /// Returns `None` if the pointer is null, or else returns a shared reference to
     /// the value wrapped in `Some`.
     ///
     /// # Safety
     ///
-    /// While this method and its mutable counterpart are useful for
-    /// null-safety, it is important to note that this is still an unsafe
-    /// operation because the returned value could be pointing to invalid
-    /// memory.
-    ///
     /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
     /// all of the following is true:
-    /// - it is properly aligned
-    /// - it must point to an initialized instance of T; in particular, the pointer must be
-    ///   "dereferenceable" in the sense defined [here].
+    ///
+    /// * The pointer must be properly aligned,
+    ///
+    /// * it must be "dereferencable" in the sense defined in [the module documentation], and
```

Comment:
You have a ", and" here but not on the first item. Given that the list starts with "all of the following must be true", I don't think the ", and" is necessary, but if you want to use it please do so consistently.

Reply:
The " and" was a copy-paste misstake, thanks!

Reply:
Basically these should be either self-contained sentences (starting with a capital letter and ending in a `.`), or forma  long sentence together. But the last gets kind of awkward given that the last item is multiple sentences long.

Reply:
Is there a guideline for writing safety-sections somewhere? It seems that there are several approaches, even in `std`. A few examples:
- [`str::get_unchecked`](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.get_unchecked) uses ";" and "."
- [`slice::from_raw_parts`](https://doc.rust-lang.org/nightly/core/slice/fn.from_raw_parts.html) or [`str::slice_unchecked`](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.slice_unchecked) uses "."
- [`GlobalAlloc::realloc`](https://doc.rust-lang.org/nightly/core/alloc/trait.GlobalAlloc.html#method.realloc) (and other methods from `core::alloc`) uses a mix of ", and", ",", and "."

Reply:
I don't think there is... I'd have used `.` for all of those.